File: sdk/python/apex_sdk/orchestration.py
"""Python SDK for the orchestration service."""

from __future__ import annotations

from typing import Any, Dict
from .client import APEXClient


class OrchestrationAPI:
    def __init__(self, client: APEXClient) -> None:
        self._client = client

    async def run_flow(self, flow_path: str, inputs: Dict[str, Any] | None = None) -> Dict[str, Any]:
        req = {
            "flow_path": flow_path,
            "inputs": inputs or {},
        }
        res = await self._client.post(self._client.orchestration_url, "/flows/run", req)
        res.raise_for_status()
        return res.json()["context"]

File: sdk/typescript/src/orchestration.ts
export async function runFlow(flowPath: string, inputs: any = {}, baseUrl?: string) {
  const url = `${baseUrl || process.env.NEXT_PUBLIC_ORCHESTRATION_URL || ''}/flows/run`;
  const res = await fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ flow_path: flowPath, inputs })
  });
  const data = await res.json();
  if (!res.ok) throw new Error(data.detail || 'Failed to run flow');
  return data.context;
}